apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.cs

```diff
@@ -185,6 +185,33 @@ public string Sign(string bucket, string objectName, TimeSpan duration, HttpMeth
             GetEffectiveSigner(GaxPreconditions.CheckNotNull(options, nameof(options)).SigningVersion).Sign(
                 GaxPreconditions.CheckNotNull(requestTemplate, nameof(requestTemplate)), options, _blobSigner, _clock);
 
+        /// <summary>
+        /// Signs the given post policy. The result can be used to make form posting requests matching the conditions
+        /// set in the post policy.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Signing post policies is not supported by <see cref="SigningVersion.V2"/>. A <see cref="NotSupportedException"/>
+        /// will be thrown if an attempt is made to sign a post policy using <see cref="SigningVersion.V2"/>.
+        /// </para>
+        /// <para>
+        /// When a <see cref="UrlSigner"/> is created with a service account credential, the signing can be performed
+        /// with no network access. When it is created with an implementation of <see cref="IBlobSigner"/>, that may require
+        /// network access or other IO. In that case, one of the asynchronous methods should be used when the caller is
+        /// in a context that should not block.
+        /// </para>
+        /// <para>
+        /// See https://cloud.google.com/storage/docs/xml-api/post-object for more information on signed post policies.
+        /// </para>
+        /// </remarks>
+        /// <param name="postPolicy">The post policy to signed and that will be enforced when making the post request.
+        /// Must not be null.</param>
+        /// <param name="options">The options used to generate the signed post policy. Must not be null.</param>
+        /// <returns>The signed post policy, which contains all the fields that should be including in the form to post.</returns>
+        public SignedPostPolicy Sign(PostPolicy postPolicy, Options options) =>
```

Comment:
For other languages, we have a consistent method name `GenerateSignedPostPolicyV4`, looking at the design and how it is used similarly to SignedURLs, this may not be possible but wanted to raise it.

Reply:
I would prefer to keep it consistent with the existing URL signing method names if that's allright? @jskeet any preferences?

Reply:
Understood, I agree language consistency is better than across the other 7 languages.

Reply:
I'm fine with either just `Sign` or `SignPostPolicy` (the latter being reasonable as a "extra specificity" given that this is a **URL**-signer, but being asked to sign something other than a URL). But I definitely prefer both of those over `GenerateSignedPostPolicyV4`.